meson.build

```diff
@@ -137,3 +137,21 @@ configure_file(
 
 subdir('src')
 subdir('examples')
+
+# Use clang-tidy to modernize code and check for common code style issues
+# meson keeps "fixing" the command, so we can't write "\." to escape the dot in the regex
+# or use any character classes. We want to match any path starting with "../src/".
+clangtidy = find_program('run-clang-tidy', required: false)
+if clangtidy.found()
+        run_target('clangtidy-autofix',
+                command : ['run-clang-tidy', '-p', meson.build_root(),
+                        '-header-filter', '^.{2}/src/.*',
```

Comment:
But I need the location of the compile_commands.json in the `-p` parameter, which is in the build root and has all file parts relative to that. So that part is actually right!?

The second issue is to have a regex, which matches these relative paths to the source_root. I just don't know how to do this.

Reply:
use meson.build_root() where you need the build dir and meson.source_root() where you need the source directory. You can do things like `meson.source_root() + 'src/*.h'

Reply:
But this doesn't work, because it'll add the full path instead of the relative one. I specifically don't need the full path, but the path to the source file as it is stated in the compile_commands.json file, which is generated by meson/ ninja.

Reply:
Ok, let's leave it as is then and I can try to figure out later an alternative

Reply:
`@SOURCE_ROOT@` seems to give me a relative path. Still need to find a solution, that the `../` needs to be `\.\./` for the regex (but not today).

While we're at it, is `meson.current_build_dir()` or `meson.project_build_root()` the right path for the build directory? Because `meson.build_root()` is deprecated since 0.56.0.

Reply:
The only quick and not-so-hacky solution I came up with, was a small wrapper shell script, which manipulates the arguments to `run-clang-tidy` before calling it. Since the build root output by meson should always be an absolute path, this shouldn't interfere with the `-p`, but only with the `-header-filter` part.

Reply:
small wrapper script is fine